### PR TITLE
Mark deprecated protobuf p2p msgs

### PR DESCRIPTION
**Pull request by terencechain**
Part of #3147 

This PR marks the existing p2p messages and topics as deprecated

### CODE REVIEW

File: proto/beacon/p2p/v1/messages.proto [hunk 1]
```diff
@@ -11,17 +11,17 @@ import "github.com/gogo/protobuf/gogoproto/gogo.proto";
...
 enum Topic {
   UNKNOWN = 0;
-  BEACON_BLOCK_ANNOUNCE = 1;
+  BEACON_BLOCK_ANNOUNCE = 1 [deprecated=true];
   BEACON_BLOCK_REQUEST = 2;
-  BEACON_BLOCK_REQUEST_BY_SLOT_NUMBER = 3;
+  BEACON_BLOCK_REQUEST_BY_SLOT_NUMBER = 3 [deprecated=true];
   BEACON_BLOCK_RESPONSE = 4;
   BATCHED_BEACON_BLOCK_REQUEST = 5;
   BATCHED_BEACON_BLOCK_RESPONSE = 6;
```
Comment (prestonvanloon) [inline proto/beacon/p2p/v1/messages.proto]: All of this is deprecated

File: proto/beacon/p2p/v1/messages.proto [hunk 2]
```diff
@@ -32,6 +32,7 @@ message Envelope {
 }
```
Comment (prestonvanloon) [inline proto/beacon/p2p/v1/messages.proto]: Envelop is deprecated

Comment (prestonvanloon) [reply proto/beacon/p2p/v1/messages.proto]: Actually, lets leave it. If we support a protobuf encoding, we'll use this.

File: proto/beacon/p2p/v1/messages.proto (hunk 1 above)
Comment (terencechain) [reply proto/beacon/p2p/v1/messages.proto]: Ah ok. Wasn't sure if they were block req/resp were still valid. 

Deprecated everything

**Pull request merged**